src/main/java/io/jenkins/plugins/intotorecorder/transport/Grafeas.java

```diff
@@ -0,0 +1,177 @@
+/**
+ *
+ */
+package io.jenkins.plugins.intotorecorder.transport;
+
+import io.github.in_toto.models.Link;
+import io.github.in_toto.models.Artifact.ArtifactHash;
+import io.github.in_toto.keys.Signature;
+import java.net.URI;
+import org.apache.http.client.utils.URIBuilder;
+import java.net.URISyntaxException;
+import java.util.*;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+
+
+public class Grafeas extends Transport {
+
+    URI uri;
+    GrafeasOccurrence occurrence;
+
+    public class GrafeasInTotoMetadata {
+        public ArrayList<Signature> signatures = new ArrayList<Signature>();
+        public GrafeasInTotoLink signed;
+
+        public class GrafeasInTotoLink {
```

Comment:
I'm not super sure this needs to be public, considering you have that "factory" method down there to instantiate the class.